### ISSUE TITLE
Support caching for async queries

### ISSUE THREAD

**Issue by kevinwcyu** [grafana/grafana-async-query-data-js#14]
See the [issue description for cloudwatch](https://github.com/grafana/grafana/issues/63898) for some background info. That issue affects redshift and athena when going through the async query flow. This is because we bypass cache in [DataSourceWithAsyncBackend](https://github.com/grafana/grafana-async-query-data-js/blob/94b40b849b34eef684a6c8f282cb2d71a6dd1723/src/DatasourceWithAsyncBackend.ts#L142) when performing an async query.

**Comment by iwysiu** [comment]
It's currently not being cached because when we implemented the async queries, we didn't want to cache when we sent a response for a still running query to the frontend. (In async queries the frontend polls periodically until it gets a done response) If we cached a still running response, then the query would never finish. It depends on how the code is laid out, but the solution would probably be to only cache "done" responses with actual data.
